code/modules/projectiles/projectile.dm

```diff
@@ -28,6 +28,9 @@
 	var/spread = 0			//amount (in degrees) of projectile spread
 	var/legacy = 0			//legacy projectile system
 	animate_movement = 0	//Use SLIDE_STEPS in conjunction with legacy
+	var/ricochets = 0

```

Comment:
this is never set, have it go up by one every time it ricochets, and then when there's the check for if the thing it's bumping is its firer, simply ignore the firer if its ricochet (this means you can't hit yourself normally, but can ricochet things into yourself)